### DISCUSSION TITLE
Relation between ApplicationUser model and other models

### DISCUSSION THREAD

**Discussion by TabrizHabiyev** [jasontaylordev/CleanArchitecture#774] [General]
ApplicationUser model is located in Infrastructure layer. Other models are in the Domain layer.We know that the Domain layer has no access to the Infrastructure layer. 
Question: What should we do if we need a relation between a model in the domain layer and ApplicationUser?

**Comment by DJJones66** [comment]
I've been going through this myself looking at how to gain the same kind of functionality.  Just seems odd to me that this has been overlooked.

**Comment by prince272** [comment]
Sometimes, we also need to overlook things as well to make it work.

**Comment by neozhu** [comment]
I have made it a navigation property to replated application user in my project using this clean architecture.
please refer to 
https://github.com/neozhu/CleanArchitectureWithBlazorServer/blob/main/src/Domain/Common/BaseAuditableEntity.cs

**Reply by prince272** [reply]
Oh okay that's cool

**Reply by TabrizHabiyev** [reply]
> I have made it a navigation property to replated application user in my project using this clean architecture. please refer to https://github.com/neozhu/CleanArchitectureWithBlazorServer/blob/main/src/Domain/Common/BaseAuditableEntity.cs

Yes, this is nice, but you moved the AppliationUser from the Infrastructure layer to the Domain Layer.Honestly, like you, I did it this way as a last resort. I was just wondering why @jasontaylordev kept ApplicationUser in the Infrastructure layer

**Comment by hsandhar** [comment]
My take on this.

- Application is using Microsoft.AspNetCore.Identity provider for auth and to keep things simple, especially knowing there is no explicit relationship between a TodoItem/TodoItemList and the user and no other domain specific props for that user, there was no need to add/declare anything in the Domain project. 
- If there was some type of relationship in in the domain between lists/todos and some domain specific user props then yes, domain would need to know more about this user (or some type of a navigational prop scenario), but in the context of this template/demo, the only thing tying these auditable entities back to the current user is `CreatedBy` & `ModifiedBy` which are just `string?`.
- More or less, there is nothing going on in the domain/app projects where a TodoItem/List need to know anything about the current user.
